Neuro/Tasks/TasksHandler.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections;
+using Neuro.DependencyInjection;
+using Neuro.Utilities;
+using Reactor.Utilities.Attributes;
+using UnityEngine;
+
+namespace Neuro.Tasks;
+
+[RegisterInIl2Cpp]
+public class TasksHandler : MonoBehaviour, ITasksHandler
+{
+    public TasksHandler(IntPtr ptr) : base(ptr) { }
+
+    public IContextProvider Context { get; set; }
+
+    // TODO: Move this to movement class
+    public Vector2[] CurrentPath { get; set; } = Array.Empty<Vector2>();
+    public int PathIndex { get; set; } = -1;
+
+    public IEnumerator EvaluatePath(NormalPlayerTask initial)
+    {
+        CurrentPath = Context.PathfindingHandler.FindPath(PlayerControl.LocalPlayer.transform.position, initial.Locations.At(0));
+        PathIndex = 0;
+
+        while (true)
+        {
+            yield return new WaitForSeconds(1);
+
+            PlayerTask task = PlayerControl.LocalPlayer.myTasks.At(0);
+
+            // TODO: Get the nearest location from all tasks instead of the first location of the next task
+            PlayerTask nextTask = null;
+            if (task.IsComplete)
+            {
+                Info("Task is complete");
+                foreach (PlayerTask t in PlayerControl.LocalPlayer.myTasks)
+                    if (!t.IsComplete && t.HasLocation)
+                    {
+                        nextTask = t;
+                        Info(nextTask.name);
+                        break;
+                    }
+            }
+            else
+            {
+                nextTask = task;
+            }
+
+            if (nextTask != null)
+            {
+                Info("Next task isn't null");
+                CurrentPath = Context.PathfindingHandler.FindPath(PlayerControl.LocalPlayer.transform.position, nextTask.Locations.At(0));
+                PathIndex = 0;
+
+                //pathfinding.DrawPath(currentPath);
+            }
+        }
+    }
+
+    private void Update()
+    {
+        if (Minigame.Instance) return;
+
+        foreach (PlayerTask task in PlayerControl.LocalPlayer.myTasks)
+        {
+            if (task.Locations == null || task.IsComplete) continue;
+
+            // TODO: Invoke Console.Use directly
+            foreach (Vector2 location in task.Locations)
+            {
+                if (Vector2.Distance(location, PlayerControl.LocalPlayer.transform.position) < 0.8f)
```

Comment:
resharper suggests inverting this if statement to avoid nesting.

Reply:
This piece of code will be altered entirely as part of this pull request and the TODO comment.